public/app/core/localisation.tsx

```diff
@@ -1,17 +1,21 @@
 import { I18n, i18n } from '@lingui/core';
 import { I18nProvider as LinguiI18nProvider } from '@lingui/react';
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 
-import { messages } from '../../locales/en/messages';
+import config from 'app/core/config';
 
 let i18nInstance: I18n;
 
-export function getI18n(locale = 'en') {
+export async function getI18n(locale = 'en') {
   if (i18nInstance) {
     return i18nInstance;
   }
-
-  i18n.load(locale, messages);
+  // Dynamically load the messages for the user's locale
+  const imp = await import(`../../locales/${locale}/messages`).catch((err) => {
+    // TODO: Properly return an error if we can't find the messages for a locale
+    return err;
+  });
```

Comment:
@joshhunt Do you think this should also be behind the feature toggle in case it has some performance issues?

Reply:
Hmm is this for the case with the feature flag off? In that scenario I'm imagining we can just keep it always displaying english messages

Reply:
Also makes sense :)